tests/azure_iot_ut.c

```diff
@@ -0,0 +1,172 @@
+                      eAzureIoTSuccess );   
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesFind_Failure( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+    const uint8_t * pucOutValue;
+    uint32_t ulOutValueLength;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                      ucBuffer, 0, sizeof( ucBuffer ) ),
+                      eAzureIoTSuccess );
+
+    assert_int_equal( AzureIoT_MessagePropertiesAppend( &xTestMessageProperties,
+                                                        ucTestKey, sizeof( ucTestKey ) - 1,
+                                                        ucTestValue, sizeof( ucTestValue ) - 1 ),
+                      eAzureIoTSuccess );
+
+    /* Failed for NULL key */
+    assert_int_not_equal( AzureIoT_MessagePropertiesFind( &xTestMessageProperties,
+                                                          NULL, 0, &pucOutValue, &ulOutValueLength ),
+                          eAzureIoTSuccess );
+
+    /* Failed for NULL outvalue */
+    assert_int_not_equal( AzureIoT_MessagePropertiesFind( &xTestMessageProperties,
+                                                          ucTestKey, sizeof( ucTestKey ) - 1,
+                                                          NULL, 0 ),
+                          eAzureIoTSuccess ); 
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesFind_Success( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+    const uint8_t * pucOutValue;
+    uint32_t ulOutValueLength;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                      ucBuffer, 0, sizeof( ucBuffer ) ),
+                      eAzureIoTSuccess );
+
+    assert_int_equal( AzureIoT_MessagePropertiesAppend( &xTestMessageProperties,
+                                                        ucTestKey, sizeof( ucTestKey ) - 1,
+                                                        ucTestValue, sizeof( ucTestValue ) - 1 ),
+                      eAzureIoTSuccess );
+
+    assert_int_equal( AzureIoT_MessagePropertiesFind( &xTestMessageProperties,
+                                                      ucTestKey, sizeof( ucTestKey ) - 1,
+                                                      &pucOutValue, &ulOutValueLength ),
+                      eAzureIoTSuccess );
+
+    assert_int_equal( ulOutValueLength, sizeof( ucTestValue ) - 1 );
+    assert_memory_equal( pucOutValue, ucTestValue, ulOutValueLength );
+}
+/*-----------------------------------------------------------*/
+
+uint32_t ulGetAllTests()
+{
+    const struct CMUnitTest tests[] =
+    {
+        cmocka_unit_test( testAzureIoTMessagePropertiesInit_Failure ),
```

Comment:
For completeness I think we should have a `AzureIoT_Init` and `AzureIoT_Deinit` UT here, even if it's just a simple call.  This will help with any code-coverage based on UT at minimum :), and good to have extra guard rails up.